Lib/test/test_socket.py

```diff
@@ -2275,6 +2275,31 @@ def testFromFd(self):
     def _testFromFd(self):
         self.serv_conn.send(MSG)
 
+    @requireAttrs(socket.socket, "sendmsg")
+    @requireAttrs(socket, "AF_UNIX")
+    @unittest.skipIf(sys.platform == 'darwin', 'test not working for MacOSX')
+    @unittest.skipIf(sys.platform.startswith(('freebsd', 'netbsd', 'gnukfreebsd')), 'test not working for freebsd')
```

Comment:
Wait, why are we ignoring the test in FreeBSD? We should fix the code/test instead, right?

Reply:
Fixing means I remove those parameters that freebsd complains about in the function itself. Is that the right fix? and will require changing the docs too to say so. My assumption is that freebsd doesn't support those attributes we are calling that is why it is failing.

Reply:
Also I skipped MacOS for same reason, not supporting those attributes.

Reply:
FREEBSD supports those functions with `SCM_RIGHTS`:

https://www.freebsd.org/cgi/man.cgi?query=recvmsg&sektion=2&manpath=FreeBSD+6.0-RELEASE

As well as MacOSX:

https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/recvmsg.2.html

I don't think we should ignore the test in those systems.

Reply:
The error returned is general pointing to invalid attributes. It may not be SCM_RIGHTS but may be it is SOL_SOCKET that is also used in the definition of send_fds() in this case. I do not have so much time to investigate this freebsd thing so fast now given that I have to sketch for a system in the first place. So If anyone wants to pick up the investigation, well and good :)

Reply:
`SOL_SOCKET` is supported in both platforms (check the man pages for reference) but clearly there is something wrong with either the combination or that `SCM_RIGHTS` should be handled differently on both platforms, but IMHO skipping the test is not the correct thing to do here.

Reply:
I agree finding a remedy is better. I may have time next week maybe to simulate this freebsd thing to troubleshoot this but If anyone wants to jump in now, no problem :)

Reply:
I will try to find some time this week to help you in investigating this :)